### ISSUE TITLE
Malformed pom.xml prevent single module build and use as dependency

### ISSUE THREAD

**Issue by aaime** [opengeospatial/teamengine#105]
Trying to build a single teamengine module in isolation results in the following error:

{code}
aaime@hydra ~/devel/cite/engine/teamengine-web $ mvn clean install -nsu
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for org.opengis.cite.teamengine:teamengine-web:war:4.1.0
[WARNING] 'version' contains an expression but should be a constant. @ org.opengis.cite:teamengine:${project.version}, /home/aaime/devel/cite/engine/pom.xml, line 9, column 12
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building TEAM Engine - Web Application 4.1.0
[INFO] ------------------------------------------------------------------------
Downloading: https://repo.maven.apache.org/maven2/org/opengis/cite/teamengine/${project.version}/teamengine-${project.version}.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 0.825 s
[INFO] Finished at: 2015-06-21T15:06:37+02:00
[INFO] Final Memory: 10M/240M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project teamengine-web: Could not resolve dependencies for project org.opengis.cite.teamengine:teamengine-web:war:4.1.0: Failed to collect dependencies at org.opengis.cite.teamengine:teamengine-core:jar:4.1.0: Failed to read artifact descriptor for org.opengis.cite.teamengine:teamengine-core:jar:4.1.0: Could not transfer artifact org.opengis.cite:teamengine:pom:${project.version} from/to central (https://repo.maven.apache.org/maven2): Illegal character in path at index 66: https://repo.maven.apache.org/maven2/org/opengis/cite/teamengine/${project.version}/teamengine-${project.version}.pom -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
{code}

This happens because the projects use ${project.version} to identify the parent, whilst in the pom.xml itself the project.version is not set at all. To fix it, set the version manually for the parent itself, and the child module will not need further declarations.

**Comment by bermud** [comment]
@upendra-sachan @rjmartell  can you please investigate? 
Maybe the reason is that the modules have not been register in the maven repository?

**Comment by upendra-sachan** [comment]
Ok, Luis

Thanks.


------
Upendra Sachan
Mindstix Software Labs
http://www.mindstix.com

*Email:* upendra.sachan@mindstix.com <blah@mindstix.com>
*Phone:* +91 8446253174
*Skype:* upendra.sachan1

**Comment by aaime** [comment]
Hi, here is a tentative solution, following the same approach we use in GeoTools/GeoServer: https://github.com/opengeospatial/teamengine/pull/106 
The version numbers need to be modified on release of course, but it's a quick matter of running something like:

````
find . -name pom.xml -exec sed -i "s/4.1-SNAPSHOT/4.1.1/g" {} \;
````

**Comment by bermud** [comment]
@aaime, Thank you for providing this solution!

**Comment by bermud** [comment]
Fixed in master and 4.1.0 release (which will be officially release this week).
This command worked for me: 
```
find ./ -name pom.xml -exec sed -i -e 's/4.1.0/4.2-SNAPSHOT/g' {} \;
```
